### PR TITLE
added: code style convention and updated dependencies

### PR DESCRIPTION
**Pull request by andriiwp3**
- React was updated to v18
- all dependencies updated
- eslint and prettier added and configured
- the whole project was linted and formatted
- pre-commit hooks added and configured
- added props typing for all components
- .gitignore file updated
- the project documentation README.md file has been improved

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+	printWidth: 100,
```
Comment (AndriyTykhan) [inline .prettierrc.js:2]: Let's make it 120 or 160

File: .eslintrc.js [hunk 2]
```diff
@@ -1,29 +1,98 @@
...
+				alphabetize: {
+					order: 'asc',
+					caseInsensitive: true,
+				},
+			},
+		],
+		'prettier/prettier': [
+			'error',
+			{
+				printWidth: 100,
```
Comment (AndriyTykhan) [inline .eslintrc.js:57]: Let's make it 160

File: src/App.js [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
-    }
-  }, [user]);
-
-  return (
-    <div className="App">
-      <AppRouter />
-      <ToastContainer />
-    </div>
-  );
+	const { user } = useAuth0();
```
Comment (AndriyTykhan) [inline src/App.js:14]: Well, it really looks too big. Let's return spaces how they were before

**Review by AndriyTykhan — CHANGES_REQUESTED** [review]
Please return old space and quotes configuration

File: .eslintrc.js (hunk 2 above)
Comment (andriiwp3) [reply .eslintrc.js:57]: @AndriyTykhan May be better 120?

Comment (AndriyTykhan) [reply .eslintrc.js:57]: Okay, lets make it 120

**Review by AndriyTykhan — APPROVED** [review]

**Pull request merged**